.github/workflows/code_coverage.yml

```diff
@@ -0,0 +1,82 @@
+name: Code Coverage
+
+on:
+  # allows us to run workflows manually
+  workflow_dispatch:
+  pull_request:
+    paths-ignore:
+      - '.github/workflows/build_test.yml'
+      - '.github/workflows/publish_latest.yml'
+      - '.github/workflows/publish_release.yml'
+      - '.github/workflows/changelog_test.yml'
+      - 'doc/**'
+      - 'CHANGELOG.rst'
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      fail-fast: false
+      matrix:
+        ubuntu_versions : [
+          20.04,
+          22.04,
+        ]
+        pkg_mgr : [
+          apt,
+          conda,
+        ]
```

Comment:
Do you think we need to do coverage across all these variations?  What do we miss if we don't?

Reply:
I guess we probably don't.  We ignore external dependencies so the coverage should be the same across the build matrix.  They mostly are... [except](https://coveralls.io/builds/65607725) for the fact that the `20.04-apt` is covering 264 files compared to 266 files in the other 3 builds so it is slightly different 🧐.

The only downside I see to performing coverage on the entire matrix is resource usage, but we don't have any limits since our repos are public.  Do you think we should slim down the testing matrix for this workflow?

Reply:
It could be confusing to get 4 coverage results and then decide which matter?

Reply:
Hmm... I may have been misinterpreting what Coveralls meant by a [parallel build](https://docs.coveralls.io/parallel-builds).  I thought it was intended for a testing matrix like ours, but it appears that it might be intended for multiple stages of a build rather than multiple versions of the same build.  That, along with the results I'm seeing in the coverage reports (ach of the 4 builds is around 32% but the collective coverage is 37% ???) leads me to agree with you.  Which build do you want to keep?

Reply:
I don't think it matters which we keep - flip a coin or two?

Reply:
I see you did already